src/NuGet.Core/NuGet.Packaging/PackageExtraction/NuGetPackageUtils.cs

```diff
@@ -282,29 +282,7 @@ private static bool ShouldInclude(string fullName, string nupkgFileName, string
                     continue;
                 }
 
-                if (Path.GetFileName(targetFile).Length == 0)
-                {
-                    Directory.CreateDirectory(targetFile);
-                }
-                else
-                {
-                    var targetEntryPath = Path.GetDirectoryName(targetFile);
-                    if (!Directory.Exists(targetEntryPath))
-                    {
-                        Directory.CreateDirectory(targetEntryPath);
-                    }
-
-                    using (var entryStream = entry.Open())
-                    {
-                        using (var targetStream
-                            = new FileStream(targetFile, FileMode.Create, FileAccess.Write, FileShare.None))
-                        {
-                            entryStream.CopyTo(targetStream);
-                        }
-                    }
-
-                    File.SetLastWriteTimeUtc(targetFile, entry.LastWriteTime.UtcDateTime);
-                }
+                await PackageHelper.CreatePackageFileAsync(targetFile, entry, CancellationToken.None);
```

Comment:
You could declare this method to not be async and return the result of `PackageHelper.CreatePackageFileAsync(targetFile, entry, CancellationToken.None);`